File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        
        int size = arr.length;
        int result = 0;
        
        for(int i = 0 ; i < size; i++){
            
            int endingArr = i + 1;
            int startArray = size-i;
            
            int totalSub = endingArr * startArray;
            int oddSub = (int)Math.ceil(totalSub / 2);
            
            if(totalSub % 2 == 1){
               oddSub ++;
            }
             result += oddSub * arr[i];
        }
        
        return result;
    }
}

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var sumOddLengthSubarrays = function(arr) {
  
    let sum = 0;
    let size = arr.length;
    
    for(let i = 0 ; i<size; i++){
        let total = (i+1)*(size-i)
        sum += Math.ceil(total/2) * arr[i]
    }
    
    return sum
    
};